File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main(void) {
	int result=0;
	stack<char> s;
	string str;
	cin >> str;
	for (int i = 0; i < str.length(); i++) {
		if (str[i] == '(') {
			s.push(str[i]);
		}
		else {
			s.pop();
			if (str[i-1] == '(') {
				result += s.size();
			}
			else {
				result++;
			}
		}
	}
	cout << result;
	return 0;
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws Exception{
        int answer = 0;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<String> st = new Stack<>();
        String[] s = br.readLine().split("");
        for(int i=0; i<s.length; i++){
            if("(".equals(s[i])){
                st.push(s[i]);
            }else{
                st.pop();
                if("(".equals(s[i-1])){
                    answer+=st.size();
                }else{
                    answer+=1;
                }
            }


        }
        System.out.println(answer);
        // ()(
    }
}